### ISSUE TITLE
Potential error in calling scaled_dot_product_attention in  pytorch/torch/testing/_internal/distributed/_tensor /common_dtensor.py ??

### ISSUE THREAD

**Issue by githubsgi** [pytorch/pytorch#151878]
### 🐛 Describe the bug

Wondering whether the following call  at [here ](https://github.com/pytorch/pytorch/blob/a02eae8142ddd8fbf068a3e17fc0dd276d92fc78/torch/testing/_internal/distributed/_tensor/common_dtensor.py#L155) is  correct .   The mask is set to None and use_attn_mask is set to True. 

`

        output = F.scaled_dot_product_attention(
            queries,
            keys,
            values,
            None,
            self.dropout_p if self.training else 0,
            self.use_attn_mask,
        )

`

Per this documentation [here ](https://github.com/pytorch/pytorch/blob/40cf49d4607cf59453193321986eac34a8fbaa93/torch/nn/functional.py#L5806) and [here ](https://github.com/pytorch/pytorch/blob/40cf49d4607cf59453193321986eac34a8fbaa93/aten/src/ATen/native/transformers/attention.cpp#L718) is no use_attn_mask argument for scaled_dot_product_attention.   

### Versions

version is master , but does not matter here.

**Comment by XilunWu** [comment]
> Per these documentations, is no use_attn_mask argument for scaled_dot_product_attention.

I think that corresponds to `is_causal` arugment, i.e. `is_causal=self.use_attn_mask`.

**Comment by githubsgi** [comment]
@XilunWu , it is a bit confusing what exactly was the intent and what actually is happening.  It is also puzzling that this function call is actually going through, not failing with a stacktrace !
